### PR TITLE
Add column mapping to Trainer

### PR DESCRIPTION
**Pull request by lewtun**
This PR allows users to specify the column mapping of their datasets to the `(text, label)` schema expected by the `SetFitTrainer`.

I also took the liberty to add some basic logging and tests :)

## Usage

Suppose one has a dataset that doesn't have `text` and/or `label` columns. In this case, one can do the following:

```python
from datasets import load_dataset
from sentence_transformers.losses import CosineSimilarityLoss

from setfit import SetFitModel, SetFitTrainer

# Load a dataset from the Hugging Face Hub
# Columns: ['idx', 'sentence', 'label']
dataset = load_dataset("sst2")

# Simulate the few-shot regime by sampling 8 examples per class
num_classes = 2
train_ds = dataset["train"].shuffle(seed=42).select(range(8 * num_classes))
test_ds = dataset["test"].select(range(2))

# Load SetFit model from Hub
model = SetFitModel.from_pretrained("sentence-transformers/paraphrase-mpnet-base-v2")

# Create trainer
trainer = SetFitTrainer(
    model=model,
    train_dataset=train_ds,
    eval_dataset=test_ds,
    loss_class=CosineSimilarityLoss,
    batch_size=16,
    num_iterations=20, 
    column_mapping={"sentence":"text", "label": "label"} # Map dataset columns to ones expected by trainer
)

# Train!
trainer.train()
```

## TODO

- [x] Add unit tests

### CODE REVIEW

**Comment by danielkorat** [comment]
Hey @lewtun , looks great !
Thank you

**Review by danielkorat — APPROVED** [review]
